ci/e2e/run-tests.yml

```diff
@@ -27,31 +29,33 @@ run:
     set -e -x -u
 
     source dispatch/ci/e2e/config-k8s-env.sh
-
-    export NODE_IP=$(cat cluster/metadata | jq -r '.nodeIP')
     export INSTALL_DISPATCH=0
     export CI=true
     export TERM=linux
-    export INGRESS_PORT=$(kubectl get svc/ingress-nginx-ingress-controller -n kube-system -o json | jq -r '.spec.ports[1].nodePort')
-
-    export API_GATEWAY_HTTP_PORT=$(kubectl get svc/api-gateway-kongproxy -n kong -o jsonpath='{.spec.ports[?(@.port==80)].nodePort}')
-    export API_GATEWAY_HTTPS_PORT=$(kubectl get svc/api-gateway-kongproxy -n kong -o jsonpath='{.spec.ports[?(@.port==443)].nodePort}')
 
-    echo "${NODE_IP} dev.dispatch.vmware.com api.dev.dispatch.vmware.com" >> /etc/hosts
     cp dispatch-cli/dispatch /usr/local/bin/dispatch
-
     mkdir -p ~/.dispatch
-    cat << EOF > ~/.dispatch/config.json
-    {
-        "host": "dev.dispatch.vmware.com",
-        "port": ${INGRESS_PORT},
-        "organization": "",
-        "cookie": "",
-        "skipauth": true,
-        "insecure": true,
-        "Json": false
-    }
-    EOF
+
+    if [[ -n ${GKE_PROJECT_ID} ]]; then
+      export LOADBALANCER_IP=$(kubectl get svc/ingress-nginx-ingress-controller -n kube-system -o json | jq -r '.status.loadBalancer.ingress[0].ip')
+      export API_GATEWAY_IP=$(kubectl get svc/api-gateway-kongproxy -n kong -o json | jq -r '.status.loadBalancer.ingress[0].ip')
+      cp dispatch/ci/e2e/configs/dispatch-config-gke.json ~/.dispatch/config.json
+      sed -i "s/LOADBALANCER_IP/$LOADBALANCER_IP/g" ~/.dispatch/config.json
+
+      export API_GATEWAY_HTTPS_HOST="https://${API_GATEWAY_IP}:443"
+      export API_GATEWAY_HTTP_HOST="http://${API_GATEWAY_IP}:80"
+
+    else
+      export INGRESS_PORT=$(kubectl get svc/ingress-nginx-ingress-controller -n kube-system -o json | jq -r '.spec.ports[1].nodePort')
+      export NODE_IP=$(cat cluster/metadata | jq -r '.nodeIP')
+      export API_GATEWAY_HTTP_PORT=$(kubectl get svc/api-gateway-kongproxy -n kong -o jsonpath='{.spec.ports[?(@.port==80)].nodePort}')
+      export API_GATEWAY_HTTPS_PORT=$(kubectl get svc/api-gateway-kongproxy -n kong -o jsonpath='{.spec.ports[?(@.port==443)].nodePort}')
+      echo "${NODE_IP} dispatch.local dev.dispatch.local" >> /etc/hosts
```

Comment:
Nit: We can directly use the NODE_IP in API_GATEWAY_HTTP(S)_HOST vars.  No need to fix in this patch unless you revise it.

Reply:
Good point, we don't need it anymore. If we need to e2e restore local clusters, I will change it.